lfg/parser.js

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env node
+
+'use strict';
+const Parser = require('../.lib/parser.js');
+
+/**
+ * Recognizes the accesses to the platform Le Figaro
+ * @param  {Object} parsedUrl an object representing the URL to analyze
+ *                            main attributes: pathname, query, hostname
+ * @param  {Object} ec        an object representing the EC whose URL is being analyzed
+ * @return {Object} the result
+ */
+module.exports = new Parser(function analyseEC(parsedUrl, ec) {
+  let result = {};
+  let path = parsedUrl.pathname;
+  // uncomment this line if you need parameters
+  let param = parsedUrl.query || {};
+
+  // use console.error for debuging
+  // console.error(parsedUrl);
+
+  let match;
+
+  if ((match = /^\/reader\/[a-z0-9-]+$/i.exec(path)) !== null) {
+    // /reader/d6a0444a-897e-43d6-89f9-a971c71a99a6?origin=%2Fcatalog%2Fle-figaro%2Fle-figaro%2F2022-04-27
+    result.rtype = 'ISSUE';
+    result.mime = 'HTML';
+    result.unitid = param.origin;
+    let date;
+    if ((date = /([0-9]+-[0-9]{2}-[0-9]{2})$/i.exec(param.origin)) !== null) {
+      result.publication_date = date[1];
+    }
+
+  } else if (/^\/recherche\/(?!\/).+\/?$/i.test(path)) {
```

Comment:
The `negative lookahead` that you are using here will not behave the way you think. You should use `[^\/]+` is you want to capture all characters except slashes.

```suggestion
  } else if (/^\/recherche\/[^\/]+\/?$/i.test(path)) {
```

I encourage you to test your regular expressions (with regex101 or something similar) when you experiment with this kind of advanced regex expression